options/demo-importer/demo-files/default/theme_options.txt
###a:76:{s:8:"last_tab";s:1:"0";s:8:"mts_logo";s:0:"";s:11:"mts_favicon";s:0:"";s:14:"mts_touch_icon";s:0:"";s:14:"mts_metro_icon";s:0:"";s:20:"mts_twitter_username";s:15:"mythemeshopteam";s:14:"mts_feedburner";s:39:"http://feeds.feedburner.com/mythemeshop";s:15:"mts_header_code";s:0:"";s:18:"mts_analytics_code";s:0:"";s:23:"mts_pagenavigation_type";s:1:"1";s:15:"mts_ajax_search";s:1:"1";s:14:"mts_full_posts";s:1:"0";s:14:"mts_responsive";s:1:"1";s:7:"mts_rtl";s:1:"0";s:17:"mts_shop_products";s:1:"9";s:15:"mts_prefetching";s:1:"0";s:13:"mts_lazy_load";s:1:"1";s:20:"mts_lazy_load_thumbs";s:1:"1";s:21:"mts_lazy_load_content";s:1:"1";s:12:"mts_async_js";s:1:"1";s:21:"mts_remove_ver_params";s:1:"1";s:15:"mts_optimize_wc";s:1:"1";s:16:"mts_color_scheme";s:7:"#0274be";s:10:"mts_layout";s:8:"cslayout";s:14:"mts_background";a:10:{s:5:"color";s:7:"#eeeeee";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:6:"repeat";s:10:"attachment";s:6:"scroll";s:8:"position";s:8:"left top";s:4:"size";s:5:"cover";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}s:8:"parallax";s:1:"0";}s:14:"mts_custom_css";s:0:"";s:12:"mts_lightbox";s:1:"0";s:16:"mts_header_style";s:18:"logo_in_nav_header";s:14:"mts_sticky_nav";s:1:"1";s:20:"mts_show_primary_nav";s:1:"1";s:19:"mts_header_section2";s:1:"1";s:23:"mts_header_social_icons";s:1:"1";s:17:"mts_header_social";a:4:{i:0;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"2";s:21:"mts_header_icon_title";s:8:"Facebook";s:15:"mts_header_icon";s:8:"facebook";s:20:"mts_header_icon_link";s:32:"https://facebook.com/mythemeshop";}i:1;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"3";s:21:"mts_header_icon_title";s:7:"Twitter";s:15:"mts_header_icon";s:7:"twitter";s:20:"mts_header_icon_link";s:35:"https://twitter.com/mythemeshopteam";}i:2;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"4";s:21:"mts_header_icon_title";s:11:"Google Plus";s:15:"mts_header_icon";s:11:"google-plus";s:20:"mts_header_icon_link";s:31:"https://google.com/+MyThemeShop";}i:3;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"5";s:21:"mts_header_icon_title";s:8:"YoutTube";s:15:"mts_header_icon";s:12:"youtube-play";s:20:"mts_header_icon_link";s:40:"https://youtube.com/user/MyThemeShopTube";}}s:19:"mts_featured_slider";s:1:"1";s:23:"mts_featured_slider_cat";a:2:{i:0;s:1:"2";i:1;s:1:"4";}s:23:"mts_featured_slider_num";s:1:"3";s:16:"mts_thumb_layout";s:16:"large_home_thumb";s:23:"mts_featured_categories";a:1:{i:0;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"1";s:21:"mts_featured_category";s:6:"latest";s:30:"mts_featured_category_postsnum";s:1:"3";}}s:27:"mts_home_headline_meta_info";a:1:{s:7:"enabled";a:4:{s:6:"author";s:11:"Author Name";s:4:"date";s:4:"Date";s:8:"category";s:10:"Categories";s:7:"comment";s:13:"Comment Count";}}s:14:"mts_top_footer";s:1:"1";s:18:"mts_top_footer_num";s:1:"4";s:19:"mts_footer_bg_color";s:7:"#222222";s:21:"mts_footer_bg_pattern";s:4:"nobg";s:28:"mts_footer_bg_pattern_upload";s:0:"";s:14:"mts_copyrights";s:73:"Theme by <a href="http://mythemeshop.com/" rel="nofollow">MyThemeShop</a>";s:23:"mts_copyrights_bg_color";s:7:"#ffffff";s:22:"mts_single_post_layout";a:2:{s:7:"enabled";a:3:{s:7:"content";s:12:"Post Content";s:7:"related";s:13:"Related Posts";s:6:"author";s:10:"Author Box";}s:8:"disabled";a:1:{s:4:"tags";s:4:"Tags";}}s:26:"mts_related_posts_taxonomy";s:10:"categories";s:20:"mts_related_postsnum";s:1:"3";s:29:"mts_single_headline_meta_info";a:1:{s:7:"enabled";a:4:{s:6:"author";s:11:"Author Name";s:4:"date";s:4:"Date";s:8:"category";s:10:"Categories";s:7:"comment";s:13:"Comment Count";}}s:14:"mts_breadcrumb";s:1:"1";s:18:"mts_author_comment";s:1:"1";s:16:"mts_comment_date";s:1:"1";s:26:"mts_social_button_position";s:8:"floating";s:27:"mts_social_buttons_on_pages";s:1:"0";s:18:"mts_social_buttons";a:2:{s:7:"enabled";a:5:{s:13:"facebookshare";s:14:"Facebook Share";s:8:"facebook";s:13:"Facebook Like";s:7:"twitter";s:7:"Twitter";s:5:"gplus";s:11:"Google Plus";s:9:"pinterest";s:9:"Pinterest";}s:8:"disabled";a:2:{s:8:"linkedin";s:8:"LinkedIn";s:7:"stumble";s:11:"StumbleUpon";}}s:17:"mts_header_adcode";s:0:"";s:18:"mts_posttop_adcode";s:0:"";s:23:"mts_posttop_adcode_time";s:1:"0";s:18:"mts_postend_adcode";s:0:"";s:23:"mts_postend_adcode_time";s:1:"0";s:20:"mts_sidebar_for_home";s:0:"";s:20:"mts_sidebar_for_post";s:0:"";s:20:"mts_sidebar_for_page";s:0:"";s:23:"mts_sidebar_for_archive";s:0:"";s:24:"mts_sidebar_for_category";s:0:"";s:19:"mts_sidebar_for_tag";s:0:"";s:20:"mts_sidebar_for_date";s:0:"";s:22:"mts_sidebar_for_author";s:0:"";s:22:"mts_sidebar_for_search";s:0:"";s:24:"mts_sidebar_for_notfound";s:0:"";s:20:"mts_sidebar_for_shop";s:0:"";s:23:"mts_sidebar_for_product";s:0:"";s:19:"mts_typography_sets";a:1:{i:0;s:5:"latin";}s:29:"google_typography_collections";a:18:{i:0;a:9:{s:7:"default";b:1;s:12:"preview_text";s:4:"Logo";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"32px";s:10:"font_color";s:4:"#222";s:13:"css_selectors";s:7:"#logo a";s:14:"additional_css";s:26:"text-transform: uppercase;";}i:1;a:8:{s:7:"default";b:1;s:12:"preview_text";s:23:"Primary Navigation Font";s:13:"preview_color";s:5:"light";s:11:"font_family";s:7:"Raleway";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"13px";s:10:"font_color";s:4:"#777";s:13:"css_selectors";s:21:"#primary-navigation a";}i:2;a:9:{s:7:"default";b:1;s:12:"preview_text";s:25:"Secondary Navigation Font";s:13:"preview_color";s:4:"dark";s:11:"font_family";s:7:"Raleway";s:12:"font_variant";s:3:"700";s:9:"font_size";s:4:"16px";s:10:"font_color";s:4:"#fff";s:13:"css_selectors";s:23:"#secondary-navigation a";s:14:"additional_css";s:26:"text-transform: uppercase;";}i:3;a:8:{s:7:"default";b:1;s:12:"preview_text";s:18:"Home Article Title";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:9:"font_size";s:4:"38px";s:12:"font_variant";s:3:"300";s:10:"font_color";s:7:"#0274BE";s:13:"css_selectors";s:20:".latestPost .title a";}i:4;a:8:{s:7:"default";b:1;s:12:"preview_text";s:20:"Single Article Title";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:9:"font_size";s:4:"38px";s:12:"font_variant";s:3:"300";s:10:"font_color";s:4:"#222";s:13:"css_selectors";s:13:".single-title";}i:5;a:8:{s:7:"default";b:1;s:12:"preview_text";s:12:"Content Font";s:13:"preview_color";s:5:"light";s:11:"font_family";s:7:"Raleway";s:9:"font_size";s:4:"16px";s:12:"font_variant";s:3:"500";s:10:"font_color";s:7:"#444444";s:13:"css_selectors";s:4:"body";}i:6;a:9:{s:7:"default";b:1;s:12:"preview_text";s:18:"Sidebar Title Font";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"20px";s:10:"font_color";s:7:"#222222";s:13:"css_selectors";s:19:"#sidebar .widget h3";s:14:"additional_css";s:26:"text-transform: uppercase;";}i:7;a:8:{s:7:"default";b:1;s:12:"preview_text";s:12:"Sidebar Font";s:13:"preview_color";s:5:"light";s:11:"font_family";s:7:"Raleway";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"16px";s:10:"font_color";s:7:"#444444";s:13:"css_selectors";s:16:"#sidebar .widget";}i:8;a:9:{s:7:"default";b:1;s:12:"preview_text";s:17:"Footer Title Font";s:13:"preview_color";s:4:"dark";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"18px";s:10:"font_color";s:7:"#ffffff";s:13:"css_selectors";s:18:".footer-widgets h3";s:14:"additional_css";s:26:"text-transform: uppercase;";}i:9;a:8:{s:7:"default";b:1;s:12:"preview_text";s:12:"Footer Links";s:13:"preview_color";s:4:"dark";s:11:"font_family";s:7:"Raleway";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"16px";s:10:"font_color";s:7:"#999999";s:13:"css_selectors";s:163:".f-widget a, footer .wpt_widget_content a, footer .wp_review_tab_widget_content a, footer .wpt_tab_widget_content a, footer .widget .wp_review_tab_widget_content a";}i:10;a:8:{s:7:"default";b:1;s:12:"preview_text";s:11:"Footer Font";s:13:"preview_color";s:4:"dark";s:11:"font_family";s:7:"Raleway";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"16px";s:10:"font_color";s:7:"#777777";s:13:"css_selectors";s:348:".footer-widgets, .f-widget .top-posts .comment_num, footer .meta, footer .twitter_time, footer .widget .wpt_widget_content .wpt-postmeta, footer .widget .wpt_comment_content, footer .widget .wpt_excerpt, footer .wp_review_tab_widget_content .wp-review-tab-postmeta, footer .advanced-recent-posts p, footer .popular-posts p, footer .category-posts p";}i:11;a:8:{s:7:"default";b:1;s:12:"preview_text";s:15:"Copyrights Font";s:13:"preview_color";s:4:"dark";s:11:"font_family";s:7:"Raleway";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"14px";s:10:"font_color";s:7:"#7e7d7d";s:13:"css_selectors";s:15:"#copyright-note";}i:12;a:8:{s:7:"default";b:1;s:12:"preview_text";s:11:"H1 Headline";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"28px";s:10:"font_color";s:7:"#222222";s:13:"css_selectors";s:2:"h1";}i:13;a:8:{s:7:"default";b:1;s:12:"preview_text";s:11:"H2 Headline";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"24px";s:10:"font_color";s:7:"#222222";s:13:"css_selectors";s:2:"h2";}i:14;a:8:{s:7:"default";b:1;s:12:"preview_text";s:11:"H3 Headline";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"22px";s:10:"font_color";s:7:"#222222";s:13:"css_selectors";s:2:"h3";}i:15;a:8:{s:7:"default";b:1;s:12:"preview_text";s:11:"H4 Headline";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"20px";s:10:"font_color";s:7:"#222222";s:13:"css_selectors";s:2:"h4";}i:16;a:8:{s:7:"default";b:1;s:12:"preview_text";s:11:"H5 Headline";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"18px";s:10:"font_color";s:7:"#222222";s:13:"css_selectors";s:2:"h5";}i:17;a:8:{s:7:"default";b:1;s:12:"preview_text";s:11:"H6 Headline";s:13:"preview_color";s:5:"light";s:11:"font_family";s:11:"Roboto Slab";s:12:"font_variant";s:6:"normal";s:9:"font_size";s:4:"16px";s:10:"font_color";s:7:"#222222";s:13:"css_selectors";s:2:"h6";}}s:15:"nhp-opts-backup";s:1:"1";}###

options/js/options.js
var changeWarn = false;
jQuery(document).ready(function(){
    var History = window.History;
    function loadTabs(){
        jQuery('.nhp-opts-group-tab').hide();
        jQuery('.nhp-opts-group-tab-link-li').removeClass('active');
        if(jQuery('#last_tab').val() == ''){
            jQuery('.nhp-opts-group-tab:first').slideDown('fast');
            jQuery('#nhp-opts-group-menu li:first').addClass('active');
        }else{
            tabid = jQuery('#last_tab').val();
            jQuery('#'+tabid+'_section_group').slideDown('fast');
            jQuery('#'+tabid+'_section_group_li').addClass('active');
            if (jQuery('#'+tabid+'_section_group_li').closest('#nhp-opts-homepage-accordion').length)
                jQuery('#nhp-opts-homepage-accordion').show();
            if (jQuery('#'+tabid+'_section_group_li').closest('#nhp-opts-blog-accordion').length)
                jQuery('#nhp-opts-blog-accordion').show();
        }
    }
    jQuery('.nhp-opts-group-tab-link-a').click(function(){
        if (this.id == 'accordion_section_group_li_a') {
           jQuery('#nhp-opts-homepage-accordion').slideToggle();
           return false;
        }
        if (this.id == 'accordion_section_group_li_a_2') {
           jQuery('#nhp-opts-blog-accordion').slideToggle();
           return false;
        }
        var $this = jQuery(this);
        if ($this.parent().hasClass('active'))
            return false;
        var relid = jQuery(this).attr('data-rel');
        
        jQuery('.nhp-opts-group-tab').hide();
        jQuery('#'+relid+'_section_group').fadeIn(400);

        jQuery('.nhp-opts-group-tab-link-li').removeClass('active');
        $this.parent().addClass('active');

        jQuery("html, body").animate({ scrollTop: jQuery('#nhp-opts-header').offset().top - 48 }, 500);

        History.pushState( {tab:relid}, document.title, "themes.php?page=theme_options&tab="+relid );

        return false;
    });

    History.Adapter.bind( window,'load statechange', function( event ){
        var State = History.getState();
        var tab = State.data.tab;

        if ( typeof tab !== 'undefined' ) {
            jQuery('#'+tab+'_section_group_li_a').click();
        } else {
            loadTabs();
        }
    });
    jQuery('#nhp-opts-form-wrapper').submit(function(){
        var currentTabID = jQuery(this).find('#nhp-opts-group-menu li.active a').attr('data-rel');
        jQuery('#last_tab').val(currentTabID);
    });
    

    if(jQuery('#nhp-opts-save').is(':visible')){
        jQuery('#nhp-opts-save').delay(4000).slideUp('slow');
    }
    function addTypographyData(container, form) {
        var i = 0;
        container.find(".collections .collection").each(function() {
            form.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][preview_text]', jQuery(this).find(".preview_text").val())
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][preview_color]', jQuery(this).find(".preview_color li.current a").attr("class"))
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_family]', jQuery(this).find(".font_family").select2('val'))
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_variant]', jQuery(this).find(".font_variant").select2('val'))
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_size]', jQuery(this).find(".font_size").select2('val'))
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_color]', jQuery(this).find(".font_color").val())
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][css_selectors]', jQuery(this).find(".css_selectors").val())
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][additional_css]', jQuery(this).find(".additional_css").val())
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][backup_font]', jQuery(this).find(".backup_font").select2('val'))
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][default]', jQuery(this).attr("data-default"))
                .addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][collection_title]', jQuery(this).find(".collection_title").val());
            
            i++;
        });
    }
    
    if(jQuery('#nhp-opts-imported').is(':visible')){
        jQuery('#nhp-opts-imported').delay(4000).slideUp('slow');
    }
    
    jQuery('#nhp-opts-footer').find('#savechanges').click(function(e) {
        // add typography data via hidden fields before submitting
        if (typography_isloaded) {
            addTypographyData(jQuery('#google_typography'),jQuery('#nhp-opts-form-wrapper'));
        }
        // AJAX save
        /*
        jQuery('#savechanges').prop('disabled', true).after('<div class="spinner" id="ajax-saving"></div>');
        changeWarn = false;
        var $form = jQuery('#nhp-opts-form-wrapper');
        jQuery.post( $form.attr('action'), $form.serialize() ).done(function() {
            jQuery('#ajax-saving').remove();
            jQuery('#savechanges').prop('disabled', false).after('<div id="ajax-saved">Settings saved!</div>');
            setTimeout(function() { jQuery('#ajax-saved').fadeOut('slow', function() { jQuery('#ajax-saved').remove(); }); }, 2000);
        });
        return false;
        */
        // AJAX save end
    });

    jQuery('input, textarea, select').not('.demo_checkbox').change(function(){
        if (!changeWarn) {
          //jQuery('#nhp-opts-save-warn').slideDown('slow');
          changeWarn = true;
        }
    });

    jQuery('#nhp-opts-form-wrapper').submit(function() {
        changeWarn = false;
    });

    window.onbeforeunload = confirmExit;
    function confirmExit() {
        if (changeWarn) {
            return nhpopts.leave_page_confirm;
        }
    }
    
    jQuery('#nhp-opts-import-code-button').click(function(e){
        e.preventDefault();
        jQuery('#nhp-opts-import-code-wrapper').toggle().find('#import-code-value').val('');
    });

    jQuery('#nhp-opts-export-code-copy').click(function(e){
        e.preventDefault();
        jQuery('#nhp-opts-export-code').toggle().select();
    });
    
    // Presets
    function scrollImportLogToBottom(){
        var element = document.getElementById("importing-modal-content");
        element.scrollTop = element.scrollHeight;
    }
    jQuery('#presets .preset .import-demo-button, #presets .preset .import-demo-options-button').on('click', function(e) {
        e.preventDefault();

        var $this = jQuery(this);
        var $parent = $this.closest('.preset');
        var confirmText = nhpopts.import_opt_confirm;
        if ( $this.hasClass('import-demo-button') ) {
            confirmText = nhpopts.import_all_confirm;
        }

        var result = confirm( confirmText );
        if ( result ) {

            var data = {};
            data.action = "mts_install_demo";
            data.demo_import_id = $parent.attr("data-demo-id");
            data.nonce = $parent.attr("data-nonce");
            data.demo_import_options = '1';
            data.demo_import_content = '0';
            data.demo_import_widgets = '0';

            if ( $this.hasClass('import-demo-button') ) {
                data.demo_import_content = '1';
                data.demo_import_widgets = '1';
            }

            $this.magnificPopup({
                items: {
                    src: '#importing-overlay',
                    type: 'inline'
                },
                modal: true
            }).magnificPopup('open');

            var last_response_len = false;
            jQuery.ajax( ajaxurl, {
                data: data,
                xhrFields: {
                    onprogress: function(e) {

                        var this_response, response = e.currentTarget.response;
                        if(last_response_len === false) {

                            this_response = response;
                            last_response_len = response.length;

                        } else {

                            this_response = response.substring(last_response_len);
                            last_response_len = response.length;
                        }

                        jQuery('#importing-modal-content').append(this_response);
                        scrollImportLogToBottom();
                    }
                }
            })
            .done(function(data) {
                jQuery('#importing-modal-header h2').text(nhpopts.import_done);
                jQuery('#importing-modal-footer-info').text(nhpopts.import_done);
                jQuery('#importing-modal-footer-button').show();
            })
            .fail(function(data) {
                jQuery('#importing-modal-header h2').text(nhpopts.import_fail);
                jQuery('#importing-modal-footer-info').text(nhpopts.import_fail);
                jQuery('#importing-modal-footer-button').show();
            });
        }

        return false;
    });

    jQuery('.remove-demo-button').on('click', function(e) {
        e.preventDefault();

        var result = confirm( nhpopts.remove_all_confirm );
        if ( result ) {

            var $this = jQuery(this);
            var data = {};
            data.action = "mts_install_demo";
            data.mts_remove_demos = "1";
            data.nonce = $this.attr("data-nonce");

            $this.magnificPopup({
                items: {
                    src: '#importing-overlay',
                    type: 'inline'
                },
                modal: true
            }).magnificPopup('open');

            var last_response_len = false;
            jQuery.ajax( ajaxurl, {
                data: data,
                xhrFields: {
                    onprogress: function(e) {

                        var this_response, response = e.currentTarget.response;
                        if(last_response_len === false) {

                            this_response = response;
                            last_response_len = response.length;

                        } else {

                            this_response = response.substring(last_response_len);
                            last_response_len = response.length;
                        }

                        jQuery('#importing-modal-content').append(this_response);
                        scrollImportLogToBottom();
                    }
                }
            })
            .done(function(data) {
                jQuery('#importing-modal-header h2').text(nhpopts.remove_done);
                jQuery('#importing-modal-footer-info').text(nhpopts.remove_done);
                jQuery('#importing-modal-footer-button').show();
            })
            .fail(function(data) {
                jQuery('#importing-modal-header h2').text(nhpopts.remove_fail);
                jQuery('#importing-modal-footer-info').text(nhpopts.remove_fail);
                jQuery('#importing-modal-footer-button').show();
            });
        }

        return false;
    });
    
    jQuery('#importing-modal-footer-button').on('click', function(e) {
        jQuery(this).prop('disabled', true ).text(nhpopts.reloading_page);
        e.preventDefault();
        location.reload();
    });

    
    // Confirm import
    jQuery('#nhp-opts-import').click(function() {
        return confirm(nhpopts.import_confirm);
    });
    
    // Confirm reset
    jQuery('input[name="'+nhpopts.opt_name+'[defaults]"]').click(function() {
        return confirm(nhpopts.reset_confirm);
    });
    
    // Disallow submission by enter key
    jQuery('#nhp-opts-form-wrapper').find('input').keydown(function(event){
        if ( event.keyCode == 13 ){
            event.preventDefault();
        }
    });
    
    // Floating footer
    var $footer = jQuery('#nhp-opts-footer');
    var $bottom = jQuery('#nhp-opts-bottom');
    
    
    $footer.addClass('floating');
    jQuery(document).on('scroll', function(){
        if ($bottom.isOnScreen()) {
            $footer.removeClass('floating');
        } else {
            $footer.addClass('floating');
        }
    });
    if ($bottom.isOnScreen()) {
        $footer.removeClass('floating');
    }
    
    // Needs JS sizing when position:fixed
    var footer_padding = $footer.innerWidth() - $footer.width();
    function resizeFloatingElements() {
        var w = jQuery('#nhp-opts-form-wrapper').width();
        $footer.width(w - footer_padding);
    }
    resizeFloatingElements();
    
    var resizeTimer;
    jQuery(window).resize(function() {
        clearTimeout(resizeTimer);
        resizeTimer = setTimeout(resizeFloatingElements, 100);
    });

    // Child theme creator
    jQuery('#nhp-opts-child-button').on('click', function(e) {
        e.preventDefault();
        var elem         = jQuery(this),
            childName    = jQuery('#nhp-opts-child-name'),
            childNameVal = childName.val();

        if ( '' === childNameVal ) {
            alert( nhpopts.child_theme_name_empty );
        } else {
            jQuery.ajax({
                url: ajaxurl, 
                method: 'post',
                data: {
                    'action' : 'mts_child_theme',
                    'child_name' : childNameVal
                },
                beforeSend: function() {
                    elem.prop('disabled', true);
                    childName.prop('disabled', true);
                },
                success: function(data) {
                    elem.prop('disabled', false);
                    childName.val('').prop('disabled', false);
                    mtsUpdateChildThemesList();
                }
            });
        }
    });
    // Refresh the existing child themes list
    function mtsUpdateChildThemesList() {
        jQuery.post( ajaxurl, { action: 'mts_list_child_themes' }, function( response ) {
            if ( response ) {
                jQuery('#child-theme-list-wrap').html( response );
            }
        });
    }
    
});

jQuery(window).load(function() {
    jQuery('#savechanges').prop('disabled', false);
});

jQuery.fn.isOnScreen = function(){

    var win = jQuery(window);

    var viewport = {
        top : win.scrollTop(),
        left : win.scrollLeft()
    };
    viewport.right = viewport.left + win.width();
    viewport.bottom = viewport.top + win.height();

    var bounds = this.offset();
    bounds.right = bounds.left + this.outerWidth();
    bounds.bottom = bounds.top + this.outerHeight();

    return (!(viewport.right < bounds.left || viewport.left > bounds.right || viewport.bottom < bounds.top || viewport.top > bounds.bottom));

};

jQuery.fn.addHiddenField = function(name, value) {
    this.each(function () {
        var elem_id = name.replace(/\W/g, '-').replace(/--+/g, '-').replace(/(^-|-$)/, '');
        if (jQuery('#'+elem_id).length) {
            // elem exists, change value
            jQuery('#'+elem_id).val(value);
        } else {
            // elem doesn't exist, create
            var input = jQuery("<input>").attr("type", "hidden").attr("id", elem_id).attr("name", name).val(value);
            jQuery(this).append(jQuery(input));
        }
        
    });
    return this;
};

var fnDelay = (function(){
  var timer = 0;
  return function(callback, ms){
    clearTimeout (timer);
    timer = setTimeout(callback, ms);
  };
})();
